components/services/bat_ads/bat_ads_client_notifier_impl.h

```diff
@@ -0,0 +1,132 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_SERVICES_BAT_ADS_BAT_ADS_CLIENT_NOTIFIER_IMPL_H_
+#define BRAVE_COMPONENTS_SERVICES_BAT_ADS_BAT_ADS_CLIENT_NOTIFIER_IMPL_H_
+
+#include <cstdint>
+#include <string>
+#include <vector>
+
+#include "brave/components/brave_ads/core/ads_client_notifier.h"
+#include "brave/components/brave_ads/core/ads_client_notifier_observer.h"
+#include "brave/components/services/bat_ads/public/interfaces/bat_ads.mojom.h"
+#include "mojo/public/cpp/bindings/pending_receiver.h"
+#include "mojo/public/cpp/bindings/receiver.h"
+
+namespace bat_ads {
+
+class BatAdsClientNotifierImpl : public bat_ads::mojom::BatAdsClientNotifier {
+ public:
+  explicit BatAdsClientNotifierImpl(
+      mojo::PendingReceiver<mojom::BatAdsClientNotifier> client_notifier);
+
+  BatAdsClientNotifierImpl(const BatAdsClientNotifierImpl& other) = delete;
+  BatAdsClientNotifierImpl& operator=(const BatAdsClientNotifierImpl& other) =
+      delete;
+
+  BatAdsClientNotifierImpl(BatAdsClientNotifierImpl&& other) noexcept = delete;
+  BatAdsClientNotifierImpl& operator=(
+      BatAdsClientNotifierImpl&& other) noexcept = delete;
+
+  ~BatAdsClientNotifierImpl() override;
+
+  // Binds the receiver by consuming the pending receiver swhich was created.
+  void BindReceiver();
+
+  void AddObserver(brave_ads::AdsClientNotifierObserver* observer);
+
```

Comment:
nit: can we remove this line between AddObserver and RemoveObserver and because AddObserver and RemoveObserver do not have comments, maybe move them above BindReceiver, your thoughts?